apps/rush/src/taskRunner/TaskRunner.ts

```diff
@@ -280,10 +282,27 @@ export default class TaskRunner {
   }
 
   /**
-   * Calculate the number of packages which must be build before we reach
+   * Checks for projects that indirectly depend on themselves.
+   */
+  private _checkForCyclicDependencies(tasks: ReadonlyArray<ITask>, dependencyChain: string[]): void {
+    for (const task of tasks) {
+      if (dependencyChain.indexOf(task.name) >= 0) {
+        throw new Error('A cyclic dependency was encountered:' + os.EOL
+          + '  ' + [...dependencyChain, task.name].reverse().join('\n  -> ') + os.EOL
+          + 'Consider using the cyclicDependencyProjects option for rush.json.');
+      }
+      dependencyChain.push(task.name);
+      this._checkForCyclicDependencies([...task.dependents], dependencyChain);
```

Comment:
>...task.dependents [](start = 40, length = 18)

Same here with the spread operator.

Reply:
task.dependents is not an array

---
In reply to: [143555876](https://github.com/Microsoft/web-build-tools/pull/390#discussion_r143555876) [](ancestors = 143555876)

Reply:
But.... I forgot that @nickpape-mfst enabled the ES6 target, so that "for" loops should now work with Iterable<T>.

I can use that instead.

---
In reply to: [143556226](https://github.com/Microsoft/web-build-tools/pull/390#discussion_r143556226) [](ancestors = 143556226,143555876)